common/client/multi_node.go

```diff
@@ -545,45 +551,154 @@ func (c *multiNode[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OP
 	return n.RPC().SendEmptyTransaction(ctx, newTxAttempt, seq, gasLimit, fee, fromAddress)
 }
 
+type sendTxResult struct {
+	Err        error
+	ResultCode SendTxReturnCode
+}
+
+// broadcastTxAsync - creates a goroutine that sends transaction to the node. Returns false, if MultiNode is Stopped
+func (c *multiNode[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OPS, TX_RECEIPT, FEE, HEAD, RPC_CLIENT]) broadcastTxAsync(ctx context.Context,
+	n SendOnlyNode[CHAIN_ID, RPC_CLIENT], tx TX, txResults chan sendTxResult) bool {
+
+	ok := c.IfNotStopped(func() {
+		// Must wrap inside IfNotStopped to avoid waitgroup racing with Close
+		c.wg.Add(1)
+		go func() {
+			defer c.wg.Done()
+
+			txErr := n.RPC().SendTransaction(ctx, tx)
+			c.lggr.Debugw("Node sent transaction", "name", n.String(), "tx", tx, "err", txErr)
+			resultCode := c.classifySendTxError(tx, txErr)
+			if resultCode != Successful && resultCode != TransactionAlreadyKnown {
+				c.lggr.Warnw("RPC returned error", "name", n.String(), "tx", tx, "err", txErr)
+			}
+
+			// we expected txResults to have sufficient buffer, otherwise we are not interested in the response
+			// and can drop it
+			select {
+			case txResults <- sendTxResult{Err: txErr, ResultCode: resultCode}:
+			default:
+			}
+
+		}()
+	})
+	if !ok {
+		c.lggr.Debugw("Cannot broadcast transaction to node; MultiNode is stopped", "node", n.String())
+	}
+
+	return ok
+}
+
+// collectTxResults - reads send transaction results from the provided channel and groups them by `SendTxReturnCode.`
+// We balance the waiting time and the number of collected results. Our target is replies from 70% of nodes,
+// but we won't wait longer than sendTxSoftTimeout since the first reply to avoid waiting
+// for a timeout from slow/unhealthy nodes.
+func (c *multiNode[CHAIN_ID, SEQ, ADDR, BLOCK_HASH, TX, TX_HASH, EVENT, EVENT_OPS, TX_RECEIPT, FEE, HEAD, RPC_CLIENT]) collectTxResults(ctx context.Context, tx TX, txResults <-chan sendTxResult) (map[SendTxReturnCode][]error, error) {
+	const quorum = 0.7
```

Comment:
How did we come up with this number? In the majority of cases, we use at most 2-3 primary nodes. Usually, it's either 1 or 2. Effectively that means that you will either have to wait for all of them (not a quorum) or wait for the timeout. Seems like a good solution in theory, but what is the actual problem we are trying to solve here? Sending to a primary node feels like an eligibility operation. It might be better to wait for a single success instead, even if it's an out of sync node (although selectNode already solves that to some extend).
Additionally, `aggregateTxResults` adds more error message handling, which might come at a cost when trying to reuse the same multinode client in other chains.
Something I like a lot though, is that we unified error classification of primaries and sendonlys

Reply:
> How did we come up with this number? In the majority of cases, we use at most 2-3 primary nodes. Usually, it's either 1 or 2. Effectively that means that you will either have to wait for all of them (not a quorum) or wait for the timeout. Seems like a good solution in theory, but what is the actual problem we are trying to solve here? Sending to a primary node feels like an eligibility operation. It might be better to wait for a single success instead, even if it's an out of sync node (although selectNode already solves that to some extend).

Good point. Given how we handle collected results it might be good idea to wait for first success, unblock the caller and do sanity checks of the rest of the responses in the background. 

> Additionally, aggregateTxResults adds more error message handling, which might come at a cost when trying to reuse the same multinode client in other chains.

I thought that we are not adding more complexity as we are reusing `ClassifySendError`. Or do you mean new critical errors?

cc @prashantkumar1982